File: StopwatchApp/TLUContact/app/src/main/java/com/example/tlucontact/MainActivity.java
package com.example.tlucontact;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.tlucontact.CBNV.CBNV_Activity;
import com.example.tlucontact.DBDV.DBNV_Activity;

public class MainActivity extends AppCompatActivity {
    private Button btn_dbdv;
    private Button btn_cbnv;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        if (getSupportActionBar() != null) {
            getSupportActionBar().hide(); // Ẩn Action Bar
        }
        btn_dbdv = findViewById(R.id.btn_dbdv);
        btn_cbnv = findViewById(R.id.btn_cbnv);
        //
        btn_dbdv.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, DBNV_Activity.class);
            startActivity(intent);
        });
        btn_cbnv.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, CBNV_Activity.class);
            startActivity(intent);
        });
        }
    }

File: StopwatchApp/app/src/main/java/com/example/stopwatchapp/MainActivity.kt
package com.example.stopwatchapp

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var tvTimer: TextView
    private lateinit var btnStart: Button
    private lateinit var btnStop: Button

    private var time = 0 // Biến lưu số giây đã trôi qua
    private var isRunning = false // Trạng thái của timer
    private val handler = Handler(Looper.getMainLooper()) // Handler để cập nhật UI

    private val runnable = object : Runnable {
        override fun run() {
            if (isRunning) {
                time++
                tvTimer.text = time.toString()
                handler.postDelayed(this, 1000) // Cập nhật sau mỗi giây
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Ánh xạ View
        tvTimer = findViewById(R.id.tvTimer)
        btnStart = findViewById(R.id.btnStart)
        btnStop = findViewById(R.id.btnStop)

        // Xử lý sự kiện nút Start
        btnStart.setOnClickListener {
            if (!isRunning) {
                isRunning = true
                handler.post(runnable) // Bắt đầu cập nhật UI
            }
        }

        // Xử lý sự kiện nút Stop
        btnStop.setOnClickListener {
            isRunning = false
            handler.removeCallbacks(runnable) // Dừng cập nhật UI
        }
    }
}
